src/HttpPsrClient.php

```diff
@@ -30,9 +31,10 @@ public function sendRequest(RequestInterface $request, array $options = []);
      * @param RequestInterface[] $requests
      * @param array              $options
      *
-     * @return ResponseInterface[]
+     * @return BatchResult
      *
      * @throws Exception
+     * @throws BatchException
```

Comment:
Isn't it too long? The BatchException API contains this information (about the result).